agent/src/agent/cli/streamsets.py

```diff
@@ -27,10 +27,18 @@ def list_():
 
 
 @click.command()
-def add():
-    streamsets.manager.create_streamsets(
-        _prompt_streamsets(StreamSets(_prompt_url(), '', '', ''))
-    )
+@click.option('--url', type=click.STRING)
+@click.option('--username', type=click.STRING, default='admin')
+@click.option('--password', type=click.STRING, default='admin')
+@click.option('--agent-ext-url', type=click.STRING, default='http://anodot-agent')
+def add(url, username, password, agent_ext_url):
+    if url:
+        _validate_streamsets_url(url)
+        _validate_agent_external_url(agent_ext_url)
+        streamsets_ = StreamSets(url, username, password, agent_ext_url)
```

Comment:
we also need to call streamsets.validator.validate(streamsets_) in order to check username and password, maybe it's possible to move this call from _prompt_streamsets() here?